api/authenticator.py

```diff
@@ -705,7 +699,8 @@ def create_authentication_document(self) -> str:
             doc["service_area"] = service_area
 
         # Add the library's public key.
-        doc["public_key"] = dict(type="RSA", value=self.public_key)
+        if self.library and self.library.public_key:
```

Comment:
Should the absence of a public key here be an error?

Reply:
I think missing a public key can arise normally in the current code. 

Since we initialize the key-pair when we load a `LibraryAuthenticator` for the first time, any library without an auth mechanism won't have a keypair yet. I don't think this warrents an error, since we would have just created a key pair once an auth mechanism is added. I don't think we'll see many libraries in this state, but we will probably see a few on our dev and staging servers.

Reply:
That sounds fine. Just wanted to check, since the keys can't be null in the database and the migration should take care of any that are.

Reply:
I was looking at the wrong code for this comment. 🤦🏻 

This is mostly for mypy, since both `self.library` and `self.library.public_key` could theoretically potentially both be `None`. I'll add an error log statement in the `else` condition here, so that we get some indication if this is ever happening anywhere.